tests/_benderjs/ckeditor/static/tools.js

```diff
@@ -1204,6 +1206,72 @@
 			} else {
 				element[ eventName ]( evt );
 			}
+		},
+
+		/**
+		 * Creates a promise from the given function. It works in a similar way as a promise constructor (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise),
+		 * with support for IE8 browser.
+		 * ```js
+		 * 	bender.tools.promise( function( resolve, reject ) {
+		 * 		setTimeout( function() {
+		 * 			var timestamp;
+		 * 			try {
+		 * 				timestamp = ( new Date() ).getTime();
+		 * 			} catch ( e ) {
+		 * 				reject( e );
+		 * 			}
+		 * 			resolve( timestamp );
+		 * 		}, 5000 );
+		 * 	} )
+		 * ```
+		 *
+		 * @param {Function} fn Function which has 2 arguments `resolve` and `reject`. Resolve should be called when promise is fulfilled and reject when promise is rejected.
+		 * @returns {deferred} deferred promise, which should be in pending state. More details: (https://github.com/kriskowal/q/wiki/API-Reference#qdefer)
```

Comment:
I somehow missed this return type, it's obviously incorrect thus we disclose `Q` library under the hood. I would go even further and document this type as `Promise` - in fact, it should only behave as a polyfill for older browser rather than different Promise implementation.